### PR TITLE
CI: Add lint check on Github Actions

### PR DESCRIPTION
**Pull request by eloycoto**
This commit adds a new check on the Github Action, where a lint is
passed on all code to make sure that there  is no mistakes.

Signed-off-by: Eloy Coto <eloy.coto@acalustra.com>

### CODE REVIEW

**Review by jakub-dzon — APPROVED** [review]
Tightening the screws, aren't we?:D Very good!

**Comment by eloycoto** [comment]
yeah!

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -32,3 +31,14 @@ jobs:
       with:
         name: test-coverage
         path: coverage.html
+
+
+  lint-code:
+    name: Lint code
+    runs-on: ubuntu-latest
```
Comment (ydayagi) [inline .github/workflows/go.yml:38]: I don't know how to add a github action. So my questions are for learning. What is the reason for specifying ubuntu-latest?

File: internal/images/registry-auth.go [hunk 2]
```diff
@@ -29,7 +29,7 @@ func (r *RegistryAuth) GetAuthFileFromSecret(ctx context.Context, namespace, nam
 	}
 	authFile, found := secret.Data[".dockerconfigjson"]
 	if !found {
-		return "", errors.New(fmt.Sprintf(".dockerconfigjson not found in %s/%s Secret", namespace, name))
+		return "", fmt.Errorf(".dockerconfigjson not found in %s/%s Secret", namespace, name)
```
Comment (ydayagi) [inline internal/images/registry-auth.go:32]: is this something that golint brought up? cause i also used 'errors.New' here and there.

File: .github/workflows/go.yml (hunk 1 above)
Comment (eloycoto) [reply .github/workflows/go.yml:38]: it's the flavour that GH uses.

File: internal/images/registry-auth.go (hunk 2 above)
Comment (eloycoto) [reply internal/images/registry-auth.go:32]: yes, Errors.New without fmt.Sprintf() is ok.

File: main.go [hunk 3]
```diff
@@ -284,7 +288,9 @@ func main() {
 		}
 
 		//@TODO This is a hack to keep compatibility now, to be deleted.
-		go http.ListenAndServe(fmt.Sprintf(":%v", Config.HttpPort), h)
+		go func() {
```
Comment (ydayagi) [inline main.go:291]: just for me to understand/learn. what is the reason to switch from single line to function block?

File: .github/workflows/go.yml (hunk 1 above)
Comment (ydayagi) [reply .github/workflows/go.yml:38]: great, thanks.

File: internal/images/registry-auth.go (hunk 2 above)
Comment (ydayagi) [reply internal/images/registry-auth.go:32]: ahhhhh it's the combination of the two that it doesn't like. ok. go figure.
Thanks again. good to know.

File: main.go (hunk 3 above)
Comment (eloycoto) [reply main.go:291]: err was not checked, and lint complains about. 

I added this mainly because it's going to be deleted in the following days.

Comment (ydayagi) [reply main.go:291]: intersting. thanks.

**Review by ydayagi — APPROVED** [review]

**Pull request merged**